packages/shim-deno/src/deno/stable/variables/std.ts

```diff
@@ -1,7 +1,101 @@
 ///<reference path="../lib.deno.d.ts" />
 
-import { File } from "../classes/FsFile.js";
+function chain<T extends (...args: any[]) => Promise<any>>(
+  fn: T,
+  cleanup?: () => void,
+): T {
+  let prev: Promise<any> | undefined;
+  return function _fn(...args) {
+    const curr = (prev || Promise.resolve())
+      .then(() => fn(...args))
+      .finally(cleanup || (() => {}))
+      .then((result) => {
+        if (prev === curr) prev = undefined;
+        return result;
+      });
+    return (prev = curr);
+  } as T;
+}
 
-export const stdin: typeof Deno.stdin = new File(0);
-export const stdout: typeof Deno.stdout = new File(1);
-export const stderr: typeof Deno.stderr = new File(2);
+export const stdin: typeof Deno.stdin = {
+  rid: 0,
+  read: chain(
+    (p) => {
+      return new Promise((resolve, reject) => {
+        process.stdin.resume();
+        process.stdin.on("error", onerror);
+        process.stdin.once("readable", () => {
+          process.stdin.off("error", onerror);
+          const data = process.stdin.read(p.length) ?? process.stdin.read();
```

Comment:
I think actually it's possible that between the `process.stdin.read(p.length)` and `process.stdin.read()` calls that stdin could receive more bytes than are in the buffer. Let's resolve it in a separate PR. I'll open an issue.

Reply:
Opened https://github.com/denoland/node_deno_shims/issues/99